include/cetl/unbounded_variant.hpp

```diff
@@ -1615,122 +1627,160 @@ CETL_NODISCARD UnboundedVariant make_unbounded_variant(typename UnboundedVariant
     return UnboundedVariant(mem_res, in_place_type_t{}, list, std::forward<Args>(args)...);
 }
 
-/// \brief Performs type-safe access to the contained object.
```

Comment:
It's hard to compare b/c I have to swap order of `get`-s and `get_if`-s (otherwise clang fails (gcc was okey)).
In reality there little changes. The most important are:
- `get`-s (const and non-const) now return `ValueType` reference - similar to `variant::get`-s. Previously it was a value copy, which was not only non-consistent with `variant` but also too restrictive for "copyable" types only. Now such limitation is lifted.
- usage of `std::enable_if_t` - actual fix for the issue.